### PR TITLE
Update JavaDoc

### PR DESCRIPTION
**Pull request by sangonzal**
Updating reference documentation

### CODE REVIEW

File: src/main/java/com/microsoft/aad/msal4j/Account.java [hunk 1]
```diff
@@ -28,6 +28,10 @@
 import lombok.Setter;
 import lombok.experimental.Accessors;
 
+/**
+ * Representation of a single user account. If modifying this object, ensure it is compliant with
+ * cache persistent model
```
Comment (henrik-me) [inline src/main/java/com/microsoft/aad/msal4j/Account.java]: should this be in the public docs? That seems like an internal comment?

File: src/main/java/com/microsoft/aad/msal4j/AppMetadataCacheEntity.java [hunk 2]
```diff
@@ -31,10 +31,13 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/**
+ * Representation of application metadata.
```
Comment (henrik-me) [inline src/main/java/com/microsoft/aad/msal4j/AppMetadataCacheEntity.java]: >Representation of  [](start = 3, length = 18)

Generally not sure this makes a lot of sense. What is this metadata about?

File: src/main/java/com/microsoft/aad/msal4j/ConfidentialClientApplication.java [hunk 3]
```diff
@@ -39,6 +39,10 @@
 
 import static com.microsoft.aad.msal4j.ParameterValidationUtils.validateNotNull;
 
+/**
+ * Class to be used to acquire tokens for confidential client applications (Web Apps, Web APIs,
+ * and daemon applications).
```
Comment (henrik-me) [inline src/main/java/com/microsoft/aad/msal4j/ConfidentialClientApplication.java]: should have an aka.ms link that points to the wiki or public docs describing the confidential client

**Comment by henrik-me** [comment]
JM, would be great to get your comments on this

**Comment by henrik-me** [comment]
> // THE SOFTWARE.

nit: consider moving to the short versions of this.

---
Refers to: src/main/java/com/microsoft/aad/msal4j/UserAssertion.java:22 in 0ff475b. [](commit_id = 0ff475b46d3a5ed3a0679add708edec41e39eac4, deletion_comment = False)

**Comment by jmprieur** [comment]
@sangonzal @henrik-me : I'll have a look at it, but I don't want to be the limiting factor.
Would that work if I proposed a PR later when I have time to look at it (maybe after you have merged,  @sangonzal ) ?

File: src/main/java/com/microsoft/aad/msal4j/AppMetadataCacheEntity.java (hunk 2 above)
Comment (sangonzal) [reply src/main/java/com/microsoft/aad/msal4j/AppMetadataCacheEntity.java]: Just has foci status. Not a public API, so removing java doc comment.

File: src/main/java/com/microsoft/aad/msal4j/Account.java (hunk 1 above)
Comment (sangonzal) [reply src/main/java/com/microsoft/aad/msal4j/Account.java]: Removed comment

File: src/main/java/com/microsoft/aad/msal4j/ConfidentialClientApplication.java (hunk 3 above)
Comment (sangonzal) [reply src/main/java/com/microsoft/aad/msal4j/ConfidentialClientApplication.java]: Added links to the Java wiki in both PCA and CCA files.

**Comment by sangonzal** [comment]
> @sangonzal @henrik-me : I'll have a look at it, but I don't want to be the limiting factor.
> Would that work if I proposed a PR later when I have time to look at it (maybe after you have merged, @sangonzal ) ?

That works for me, Jean-marc. I will merge these changes in for now, and we can continue to improve the java docs incrementally.

File: src/main/java/com/microsoft/aad/msal4j/DeviceCode.java [hunk 4]
```diff
@@ -29,42 +9,46 @@
...
+ * Response returned from the STS device code endpoint containing information necessary for
+ * device code flow
+ */
 @Accessors(fluent = true)
 @Getter
 public final class DeviceCode {
 
     /**
-     *  @return user code.
+     * code which user needs to provide when authenticating at he verification URI
```
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/msal4j/DeviceCode.java]: "he" -> the

File: src/main/java/com/microsoft/aad/msal4j/IConfidentialClientApplication.java [hunk 5]
```diff
@@ -1,54 +1,33 @@
...
     /**
-     * Acquires security token from the authority.
-     *
-     * @param parameters#scopes scopes of the access request
-     * @return A {@link CompletableFuture} object representing the
-     * {@link IAuthenticationResult} of the call. It contains Access
-     * Token and the Access Token's expiration time. Refresh Token
-     * property will be null for this overload.
+     * Acquires tokens from the authority configured in the application, for the confidential client
+     * itself (in the name of no user)
```
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/msal4j/IConfidentialClientApplication.java]: "in the name of no user" sounds weird ... like it was said by Master Yoda from Start Wars ))

File: src/main/java/com/microsoft/aad/msal4j/ITokenCacheAccessContext.java [hunk 6]
```diff
@@ -1,30 +1,10 @@
...
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-// THE SOFTWARE.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 
 package com.microsoft.aad.msal4j;
 
 /**
- * Contains parameters used by the MSAL to access the cache_data.
+ * Interface representing state of token cache
```
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/msal4j/ITokenCacheAccessContext.java]: not sure about "state of the cache" - more like context of accessing cache

File: src/main/java/com/microsoft/aad/msal4j/IUserAssertion.java [hunk 7]
```diff
@@ -1,29 +1,13 @@
...
-// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-// THE SOFTWARE.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 
 package com.microsoft.aad.msal4j;
 
+/**
+ * Interface representing user credential used by downstream application in On-Behalf-Of flow
```
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/msal4j/IUserAssertion.java]: Interface representing delegated user identity and permissions used to propagate it through the request chain in On-Behalf-Of flow.

File: src/main/java/com/microsoft/aad/msal4j/TokenCacheAccessContext.java [hunk 8]
```diff
@@ -1,32 +1,15 @@
...
+// Licensed under the MIT License.
 
 package com.microsoft.aad.msal4j;
 
 import lombok.Builder;
 import lombok.Getter;
 import lombok.experimental.Accessors;
 
+/**
+ * State of token cache
```
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/msal4j/TokenCacheAccessContext.java]: not sure about "state", more access context

File: src/main/java/com/microsoft/aad/msal4j/UserAssertion.java [hunk 9]
```diff
@@ -1,30 +1,11 @@
...
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-// THE SOFTWARE.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 
 package com.microsoft.aad.msal4j;
 
 /***
- * Credential type containing an assertion representing user credential.
+ * Credential type containing an assertion representing user credential. Used as a parameter in
```
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/msal4j/UserAssertion.java]: same comment as for interface

**Pull request merged**